ci/gpu/build.sh

```diff
@@ -120,6 +120,9 @@ function install_dask {
 
 install_dask
 
+gpuci_conda_retry remove --force pyarrow arrow-cpp
+gpuci_mamba_retry install -y "pyarrow=10.0.1" "arrow-cpp=10.0.1"
```

Comment:
# TODO: Need to remove these after https://github.com/rapidsai/integration/pull/577 is merged.

Reply:
We should use the new naming: libarrow instead of arrow-cpp.
https://github.com/conda-forge/arrow-cpp-feedstock/blob/21e42a6c6a9566acebd50cc9efd6d921635204de/recipe/meta.yaml#L186